google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java

```diff
@@ -736,29 +749,51 @@ BatchState getState() {
     return state;
   }
 
-  boolean hasDocument(DocumentReference documentReference) {
-    return pendingOperations.containsKey(documentReference);
+  List<String> getPendingDocumentPaths() {
+    return FluentIterable.from(pendingOperations)
+        .transform(
+            new Function<PendingOperation, String>() {
+              @Override
+              public String apply(PendingOperation input) {
+                return input.key;
```

Comment:
You should settle on either "key" or "path". I can see an argument for key (in that our DocumentReference.getPath method is relative to the root), but we usually use path to represent the full document path in its Proto representation. I think "key" is exactly that here.

Reply:
I prefer `documentKey`, for the reasons you mentioned, and b/c it matches better with `key` from Node.